### PR TITLE
Added pylint to project and update code

### PR DESCRIPTION
**Pull request by mrmcwethy**
Lint all the libraries! #475

I tested I2C using MetroM0 with CP 2.1

### CODE REVIEW

File: adafruit_ssd1306/ssd1306.py [hunk 1]
```diff
@@ -38,72 +46,95 @@ def __init__(self, width, height, external_vcc):
...
         self.write_cmd(SET_CONTRAST)
         self.write_cmd(contrast)
 
     def invert(self, invert):
+        """invert the pixels on the display"""
         self.write_cmd(SET_NORM_INV | (invert & 1))
 
+    def write_framebuf(self):
+        """"derived class must implement this"""
+        pass
```
Comment (tannewt) [inline adafruit_ssd1306/ssd1306.py]: `raise NotImplementedError` in this case

File: adafruit_ssd1306/__init__.py [hunk 2]
```diff
@@ -1 +1,2 @@
-from adafruit_ssd1306.ssd1306 import *
+"""force classes to be loaded"""
+from adafruit_ssd1306.ssd1306 import SSD1306_SPI, SSD1306_I2C
```
Comment (tannewt) [inline adafruit_ssd1306/__init__.py]: This was a weird pattern done in some of the drivers. Instead, rename `ssd1306.py` to `adafruit_ssd1306.py` in the top level directory.

Try renaming the base class to _SSD1306 to make it private to the module. I hadn't thought of doing that before.

File: adafruit_ssd1306/ssd1306.py [hunk 3]
```diff
@@ -38,72 +46,95 @@ def __init__(self, width, height, external_vcc):
...
+                SET_NORM_INV, # not inverted
+                # charge pump
+                SET_CHARGE_PUMP, 0x10 if self.external_vcc else 0x14,
+                SET_DISP | 0x01): # on
             self.write_cmd(cmd)
         self.fill(0)
         self.show()
 
     def poweroff(self):
+        """poweroff"""
```
Comment (tannewt) [inline adafruit_ssd1306/ssd1306.py]: Please be a bit more verbose here and start the comments with a capital letter. (Applies here and elsewhere.)

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for this fix! I added a couple more things to change.

File: adafruit_ssd1306/__init__.py (hunk 2 above)
Comment (mrmcwethy) [reply adafruit_ssd1306/__init__.py]: moved and renamed ssd1306.py.  changed the name of  SSD1306 to _SSD1306

File: adafruit_ssd1306/ssd1306.py (hunk 3 above)
Comment (mrmcwethy) [reply adafruit_ssd1306/ssd1306.py]: improved the method doc string and implemented i2c poweron

File: adafruit_ssd1306/ssd1306.py (hunk 1 above)
Comment (mrmcwethy) [reply adafruit_ssd1306/ssd1306.py]: added raise NotImplemented to all abstract method.

**Comment by mrmcwethy** [comment]
I think this is ready now.  I tested using Metro M0 CP 2.1  I do not have a SPI device, so the SPI side of things could use some testing.

**Comment by mrmcwethy** [comment]
[Reference to parent issue](https://github.com/adafruit/circuitpython/issues/475)

**Review by kattni — APPROVED** [review]
Tested successfully with SPI. Required minor change, already added. It's good to go!

**Pull request merged**